app/assets/javascripts/discourse/app/components/search-menu/results/type/result.js

```diff
@@ -0,0 +1,13 @@
+import Component from "@glimmer/component";
+import DiscourseURL from "discourse/lib/url";
+import { action } from "@ember/object";
+
+export default class PostResult extends Component {
+  @action
+  onClick(event) {
+    event.preventDefault();
+    DiscourseURL.routeTo(this.args.result.url || this.args.result.path);
+    this.args.closeSearchMenu();
+    return false;
```

Comment:
What does the `return false;` accomplish? IIRC `event.preventDefault()` is the only thing we should need ?

Reply:
Also just thinking... maybe we could simplify this whole thing by letting the browser do its thing. Something like:

```js
onClick(){
  this.args.closeSearchMenu();
  // No preventDefault - allow default `<a href` behaviour
}
```

Reply:
Oh, and maybe that means we don't even need all the new JS classes. We can just do this in the template

```html
<a href="blah" {{on "click" @closeSearchMenu}}>
```
?

Reply:
Oh great idea!